tests/path.go

```diff
@@ -138,7 +141,7 @@ func (tp *TestSuite) TestInvalidPathRemainder(t *testing.T) {
 	}
 
 	_, err = api.ResolvePath(ctx, path.New("/ipld/"+nd.Cid().String()+"/bar/baz"))
-	if err == nil || !strings.Contains(err.Error(), "no such link found") {
+	if err == nil || !errors.As(err, &resolver.ErrNoLink{}) {
```

Comment:
Seems right to me. Out of curiosity did some test pick up on this or did you just figure this was a good idea?